src/Datadog.Trace/Tracer.cs

```diff
@@ -221,6 +221,35 @@ public Span StartSpan(string operationName, ISpanContext parent = null, string s
             return span;
         }
 
+        /// <summary>
+        /// Adds the <paramref name="key"/>, <paramref name="value"/> pair as a global tag to be applied to all <see cref="Span"/>s.
+        /// </summary>
+        /// <remarks><paramref name="key"/> is whitespace-trimmed before being added.</remarks>
+        /// <param name="key">The global tag key</param>
+        /// <param name="value">The global tag value</param>
+        public void AddGlobalTag(string key, string value)
+        {
+            if (string.IsNullOrWhiteSpace(key))
+            {
+                Log.Warning("Cannot create a global tag with an empty key.");
```

Comment:
This would be an instance where I think it would make sense to use LibLog, because this is relevant to the code the user owns.

Reply:
Consider throwing an `ArgumentException`.

Reply:
> This would be an instance where I think it would make sense to use LibLog, because this is relevant to the code the user owns.

Ok, I'll look at it.

Reply:
> Consider throwing an `ArgumentException`.

@lucaspimentel It doesn't throw because adding a global tag isn't a showstopper, but would be good to have more context on when Datadog tracers throw vs. log.